packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -0,0 +1,193 @@
+            plan_id: isA.string().required(),
+            token: isA.string().required()
+          }
+        },
+        response: {
+          schema: isA.object().keys({
+            subscriptionId: isA.string().required()
+          })
+        }
+      },
+      handler: async function (request) {
+        log.begin('subscriptions.createSubscription', request);
+
+        const sessionToken = request.auth.credentials;
+        const email = sessionToken.email;
+        await customs.check(request, email, 'createSubscription');
+
+        const uid = sessionToken.uid;
+        const planId = request.payload.plan_id;
+        const token = request.payload.token;
+
+        // Find the selected plan and get its product ID
+        const plans = await subhub.listPlans();
+        const selectedPlan = plans.filter(p => p.plan_id === planId)[0];
+        if (! selectedPlan) {
+          throw error.unknownSubscriptionPlan();
+        }
+        const productName = selectedPlan.product_id;
+
+        // TODO: TBD schema for paymentResult from SubHub
+        const paymentResult = await subhub.createSubscription(uid, token, planId);
+
+        // TODO: TBD what payment token rejection looks like from SubHub
+        if (! paymentResult) {
+          throw error.rejectedSubscriptionPaymentToken();
+        }
+        const subscriptionId = paymentResult.sub_id;
+
+        await db.createAccountSubscription({
+          uid,
+          subscriptionId,
+          productName,
+          createdAt: Date.now()
+        });
+
+        log.info('subscriptions.createSubscription.success', { uid, subscriptionId });
+
+        // TODO: need a metrics spec
+        request.emitMetricsEvent(
+          'subscriptions.create_subscription',
+          { uid, plan_id: planId }
+        );
+
+        return { subscriptionId };
+      }
+    },
+    {
+      method: 'POST',
+      path: '/subscriptions/updatePayment',
+      options: {
+        auth: {
+          strategy: 'sessionToken'
+        },
+        validate: {
+          payload: {
+            token: isA.string().required()
+          }
+        }
+      },
+      handler: async function (request) {
+        log.begin('subscriptions.updatePayment', request);
+
+        const sessionToken = request.auth.credentials;
+        const email = sessionToken.email;
+        await customs.check(request, email, 'updatePayment');
+
+        const uid = sessionToken.uid;
+        const token = request.payload.token;
+
+        // TODO: TBD schema for paymentResult from SubHub
+        const paymentResult = await subhub.updateCustomer(uid, token);
+
+        // TODO: TBD what payment token rejection looks like from SubHub
+        if (! paymentResult) {
+          throw error.rejectedSubscriptionPaymentToken();
+        }
+
+        log.info('subscriptions.updatePayment.success', { uid });
+
+        // TODO: need a metrics spec
+        request.emitMetricsEvent(
+          'subscriptions.update_payment',
```

Comment:
Not a massive deal, but the events we send to `emitMetricsEvent` mostly use camel case. We try to keep them consistent to make writing queries easier in Redash. There are a couple of events that use snake case, but those are exceptions that snuck through rather than the general rule.

The amplitude events, which are specified elsewhere (see `lib/metrics/amplitude.js`), *do* use snake case because that schema was defined fresh in isolation from the rest of our event architecture.